lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -774,6 +775,7 @@ def __delete_dataset(self, trans, history_id, id, purge, **kwd):
         hda = self.hda_manager.get_owned(self.decode_id(id), trans.user, current_history=trans.history)
         self.hda_manager.error_if_uploading(hda)
 
+        model.StorageMedia.refresh_all_media_credentials(hda.dataset.active_storage_media_associations, trans.app.authnz_manager, trans.sa_session)
```

Comment:
If we can't push it down below the manager level - then at very least this belongs in the manager level and not in the API level. I feel like ``purge`` and ``delete`` below should take care of these details if they are needed.

Reply:
As per conversation at https://github.com/galaxyproject/galaxy/issues/8790 cloud-based backend support is removed from this.